### PR TITLE
CLOUD-686 delete orphaned resources

### PR DESCRIPTION
**Pull request by nmarukovich**

### CODE REVIEW

File: cloud/gcp-functions/cmd/orphandResources.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+    toDelete := false
+    for _, targetPoolList := range targetPoolAggregatedList.Items {
+        for _, targetPoolItem := range targetPoolList.TargetPools {
+            zone := strings.Split(targetPoolItem.Instances[0], "/")[8]
+            instanceName := strings.Split(targetPoolItem.Instances[0], "/")[10]
+            region := strings.Split(targetPoolItem.Region, "/")[8]
+
+            if len(targetPoolItem.Instances) > 0 {
+                _, err := computeService.Instances.Get(project, zone , instanceName).Context(ctx).Do()
```
Comment (tplavcic) [inline cloud/gcp-functions/cmd/orphandResources.go:37]: space after zone and comma

File: cloud/gcp-functions/cmd/orphandResources.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                if err != nil  && strings.Contains(err.Error(), "404") {
```
Comment (tplavcic) [inline cloud/gcp-functions/cmd/orphandResources.go:38]: double space before `&&`

**Review by tplavcic — CHANGES_REQUESTED** [review]

File: cloud/gcp-functions/cmd/orphandResources.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        "fmt"
+        "net/http"
+        "context"
+        "os"
+        "strings"
+
+        "google.golang.org/api/compute/v1"
+)
+
+func CleanOrphandResources(w http.ResponseWriter, r *http.Request) {
```
Comment (egegunes) [inline cloud/gcp-functions/cmd/orphandResources.go:14]:
```suggestion
func CleanOrphanedResources(w http.ResponseWriter, r *http.Request) {
```

File: cloud/gcp-functions/cmd/orphandResources.go (hunk 1 above)
Comment (egegunes) [reply cloud/gcp-functions/cmd/orphandResources.go:37]: this and Tomislav's other comment would be fixed by running `go fmt` on this file

File: cloud/gcp-functions/cmd/orphandResources.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    if err != nil {
+            log.Fatal("Error: get targetPoolAggregatedList", err)
+    }
+
+    toDelete := false
+    for _, targetPoolList := range targetPoolAggregatedList.Items {
+        for _, targetPoolItem := range targetPoolList.TargetPools {
+            zone := strings.Split(targetPoolItem.Instances[0], "/")[8]
+            instanceName := strings.Split(targetPoolItem.Instances[0], "/")[10]
```
[the remark this answers is not in the document]
Comment (nmarukovich) [reply cloud/gcp-functions/cmd/orphandResources.go:33]: You see we should check 2 cases: 
- we don't have instances at all (empty set)
- we have instances, but they are unavailable. 
In both cases we should delete resources

File: cloud/gcp-functions/cmd/orphanedResources.go [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+	if err != nil {
+		log.Fatal("Error: get targetPoolAggregatedList", err)
+	}
+
+	toDelete := false
+	for _, targetPoolList := range targetPoolAggregatedList.Items {
+		for _, targetPoolItem := range targetPoolList.TargetPools {
+			zone := strings.Split(targetPoolItem.Instances[0], "/")[8]
+			instanceName := strings.Split(targetPoolItem.Instances[0], "/")[10]
```
Comment (egegunes) [inline cloud/gcp-functions/cmd/orphanedResources.go]: if `targetPoolItem.Instances` is empty, this line will panic in runtime. you need to check and handle it before accesing the first element

**Review by pooknull — APPROVED** [review]

**Review by tplavcic — APPROVED** [review]

**Pull request merged**